### PR TITLE
Resizing the canvas using js/css now triggers the resize event

### PR DESCRIPTION
**Pull request by yui-915**
fixes [macroquad#787](https://github.com/not-fl3/macroquad/issues/787) (I assume)

### CODE REVIEW

**Comment by InnocentusLime** [comment]
As the original issue creator, I have done some local tests and now the application **DOES** seem to be responding to me changing the canvas size in the browser! :D

I have tested this on both Chrome and Firefox. But I have no way to test it on Safari

File: js/gl.js [hunk 1]
```diff
@@ -420,13 +420,14 @@ function _webglGet(name_, p, type) {
 var Module;
 var wasm_exports;
 
-function resize(canvas, on_resize) {
+function resize(canvas, on_resize, force = false) {
```
Comment (not-fl3) [inline js/gl.js:423]: I can't see this `force` flag being used, am I mising something?

File: js/gl.js [hunk 2]
```diff
@@ -420,13 +420,14 @@ function _webglGet(name_, p, type) {
...
     var dpr = dpi_scale();
     var displayWidth = canvas.clientWidth * dpr;
     var displayHeight = canvas.clientHeight * dpr;
 
     if (canvas.width != displayWidth ||
-        canvas.height != displayHeight) {
+        canvas.height != displayHeight ||
+        force) {
```
Comment (not-fl3) [inline js/gl.js:430]: I am concerned about this `force` flag. Why do we need to send a resize message if size is still the same?

File: js/gl.js [hunk 3]
```diff
@@ -1263,6 +1264,9 @@ var importObject = {
                     wasm_exports.touch(SAPP_EVENTTYPE_TOUCHES_MOVED, touch.identifier, relative_position.x, relative_position.y);
                 }
             });
+            new ResizeObserver(function (_entries) {
```
Comment (not-fl3) [inline js/gl.js:1267]: if we use both `onresize` and `ResizeObserver`, wouldn't it generate resize message twice?

File: js/gl.js (hunk 2 above)
Comment (lenscas) [reply js/gl.js:430]: haven't looked at the code so I might be wrong

but perhaps it is a way to update the size if the CSS or similar changes?

Comment (not-fl3) [reply js/gl.js:430]: Maybe I am missing something, but I can't see how any sort of resize can be meaningful if  canvas.width/height is still the same?

File: js/gl.js (hunk 3 above)
Comment (yui-915) [reply js/gl.js:1267]: yes .... especially that `ResizeObserver` is triggered after `window.onresize`, and it has that `force` flag

simply removing `window.onresize` would solve it
but it'll cause the canvas to blink black when resizing the window, probably due to the order things are done in the browser's event loop and the fact that updating the canvas size clears it 

the best solution I found was changing `ResizeObserver` to the following:
```js
new ResizeObserver(() => {
    let dpr = dpi_scale();
    let displayWidth = Math.floor(canvas.clientWidth * dpr);
    let displayHeight = Math.floor(canvas.clientHeight * dpr);
    if (wasm_exports.resize)
        wasm_exports.resize(displayWidth, displayHeight);
}).observe(canvas);
```
and removing the `on_resize` callback from the `resize` function 
(also the `force` flag as it's no longer needed)

File: js/gl.js (hunk 2 above)
Comment (yui-915) [reply js/gl.js:430]: as far as my tests went

when forcing the canvas to be "fullscreen"
resizing the browser window will change `canvas.clientWidth`, but not `canvas.width`

but when removing `width: 100%` and `height: 100%` from the canvas css
manually updating the canvas width using js (for example by doing `canvas.width += 10;`)
will change both `canvas.width` and `canvas.clientWidth` 
so the condition evaluates to false, and on_resize method doesn't get called

same applies for height

File: js/gl.js (hunk 3 above)
Comment (yui-915) [reply js/gl.js:1267]: something like yui-915/miniquad@53b457e4

**Pull request closed without merging**